test_utils/src/lib.rs

```diff
@@ -917,3 +919,12 @@ pub fn mock_network(keypair: NetworkKeyPair, address: &Multiaddr) -> anemo::Netw
         .start(anemo::Router::new())
         .unwrap()
 }
+
+pub fn random_network() -> anemo::Network {
+    let network_key = NetworkKeyPair::generate(&mut OsRng).private().0.to_bytes();
```

Comment:
can this be implemented in terms of `mock_network`?

Also: is it too late to pedantically rename `mock_network`? It's not really a mock, I'd just call it `test_network`

Reply:
yep we can do both